src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java

```diff
@@ -533,6 +495,34 @@ public String switchToNewBrowserTab() {
     return driver.getCurrentUrl();
   }
 
+  private String getNewTab(String parentTab) {
+    Optional<String> newTab = driver
+      .getWindowHandles()
+      .stream()
+      .filter(handleName -> !handleName.equals(parentTab))
+      .findFirst();
+    return newTab.orElseThrow(() -> new NotFoundException("New tab not found!"));
+  }
+
+  private String switchToNewTab(String parentTab) {
+    String newTab = getNewTab(parentTab);
+    driver.switchTo().window(newTab);
+    return newTab;
+  }
+
+  private void waitForLinkOpenedInNewTab(WebElement link) {
+    int tabs = driver.getWindowHandles().size();

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "tabs" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)